src/commands/dev/audit/messages.ts

```diff
@@ -0,0 +1,350 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import * as fs from 'fs';
+import { join, parse, relative, resolve } from 'path';
+import { Logger, Messages } from '@salesforce/core';
+import { Flags, SfCommand } from '@salesforce/sf-plugins-core';
+import { Duration, ThrottledPromiseAll } from '@salesforce/kit';
+import { MultiDirectedGraph } from 'graphology';
+
+export type AuditResults = {
+  unusedBundles: string[];
+  unusedMessages: Array<{ Bundle: string; Name: string }>;
+  missingMessages: Array<{ File: string; Name: string; SourceVar: string; Bundle: string; IsLiteral: boolean }>;
+};
+
+type NodeType = {
+  type: 'bundle' | 'source' | 'message' | 'messageReference' | 'bundleReference';
+};
+
+type FileNode = NodeType & {
+  path: string;
+};
+
+type BundleNode = NodeType & {
+  name: string;
+};
+
+type BundleRefNode = NodeType & {
+  variable: string;
+  name: string;
+};
+
+type MessageNode = NodeType & {
+  key: string;
+};
+
+type MessageRefNode = NodeType & {
+  key: string;
+  isLiteral: boolean;
+};
+
+type Node = FileNode | BundleNode | MessageNode | MessageRefNode | BundleRefNode;
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.load('@salesforce/plugin-dev', 'audit.messages', [
+  'summary',
+  'description',
+  'examples',
+  'flags.messages-dir.summary',
+  'flags.messages-dir.description',
+  'flags.project-dir.summary',
+  'flags.project-dir.description',
+  'flags.source-dir.summary',
+  'flags.source-dir.description',
+  'missingMessagesExplanation',
+  'missingMessagesNonLiteralWarning',
+  'missingMessagesFound',
+  'noMissingMessagesFound',
+  'noUnusedBundlesFound',
+  'noUnusedMessagesFound',
+  'unusedBundlesFound',
+  'unusedMessagesFound',
+]);
+
+export default class AuditMessages extends SfCommand<AuditResults> {
+  public static summary = messages.getMessage('summary');
+  public static description = messages.getMessage('description');
+  public static examples = messages.getMessages('examples');
+
+  public static flags = {
+    'project-dir': Flags.directory({
+      summary: messages.getMessage('flags.project-dir.summary'),
+      char: 'p',
+      description: messages.getMessage('flags.project-dir.description'),
+      default: './',
+    }),
+    'messages-dir': Flags.directory({
+      summary: messages.getMessage('flags.messages-dir.summary'),
+      char: 'm',
+      description: messages.getMessage('flags.messages-dir.description'),
+      default: './messages',
+    }),
+    'source-dir': Flags.directory({
+      summary: messages.getMessage('flags.source-dir.summary'),
+
+      char: 's',
+      description: messages.getMessage('flags.source-dir.description'),
+      default: './src',
+    }),
+  };
+  private flags: { 'source-dir': string; 'messages-dir': string };
```

Comment:
You can change this to

```typescript
private flags: Interfaces.InferredFlags<typeof AuditMessages>
```

so you don't have to duplicate all the flag names and types